filters/include/pcl/filters/impl/morphological_filter.hpp

```diff
@@ -52,154 +52,156 @@
+                break;
+              }
             }
           }
         }
-      }
 
-      cloud_temp.swap (cloud_out);
+        cloud_temp.swap (cloud_out);
 
-      for (std::size_t p_idx = 0; p_idx < cloud_temp.points.size (); ++p_idx)
-      {
-        Eigen::Vector3f bbox_min, bbox_max;
-        std::vector<int> pt_indices;
-        float minx = cloud_temp.points[p_idx].x - half_res;
-        float miny = cloud_temp.points[p_idx].y - half_res;
-        float minz = -std::numeric_limits<float>::max ();
-        float maxx = cloud_temp.points[p_idx].x + half_res;
-        float maxy = cloud_temp.points[p_idx].y + half_res;
-        float maxz = std::numeric_limits<float>::max ();
-        bbox_min = Eigen::Vector3f (minx, miny, minz);
-        bbox_max = Eigen::Vector3f (maxx, maxy, maxz);
-        tree.boxSearch (bbox_min, bbox_max, pt_indices);
-
-        if (!pt_indices.empty ())
+        for (std::size_t p_idx = 0; p_idx < cloud_temp.points.size (); ++p_idx)
         {
-          Eigen::Vector4f min_pt, max_pt;
-          pcl::getMinMax3D<PointT> (cloud_temp, pt_indices, min_pt, max_pt);
-
-          switch (morphological_operator)
+          Eigen::Vector3f bbox_min, bbox_max;
+          std::vector<int> pt_indices;
+          float minx = cloud_temp.points[p_idx].x - half_res;
+          float miny = cloud_temp.points[p_idx].y - half_res;
+          float minz = -std::numeric_limits<float>::max ();
+          float maxx = cloud_temp.points[p_idx].x + half_res;
+          float maxy = cloud_temp.points[p_idx].y + half_res;
+          float maxz = std::numeric_limits<float>::max ();
+          bbox_min = Eigen::Vector3f (minx, miny, minz);
+          bbox_max = Eigen::Vector3f (maxx, maxy, maxz);
+          tree.boxSearch (bbox_min, bbox_max, pt_indices);
+
+          if (!pt_indices.empty ())
           {
-            case MORPH_OPEN:
-            default:
-            {
-              cloud_out.points[p_idx].z = max_pt.z ();
-              break;
-            }
-            case MORPH_CLOSE:
+            Eigen::Vector4f min_pt, max_pt;
+            pcl::getMinMax3D<PointT> (cloud_temp, pt_indices, min_pt, max_pt);
+
+            switch (morphological_operator)
             {
-              cloud_out.points[p_idx].z = min_pt.z ();
-              break;
+              case MORPH_OPEN:
+              default:
+              {
+                cloud_out.points[p_idx].z = max_pt.z ();
+                break;
+              }
+              case MORPH_CLOSE:
+              {
+                cloud_out.points[p_idx].z = min_pt.z ();
+                break;
+              }
             }
           }
         }
+        break;
+      }
+      default:
+      {
+        PCL_ERROR ("Morphological operator is not supported!\n");
+        break;
       }
-      break;
-    }
-    default:
-    {
-      PCL_ERROR ("Morphological operator is not supported!\n");
-      break;
     }
-  }
 
-  return;
+    return;
+  }
 }
```

Comment:
Missing `// namespace pcl` (not in the coding standard, but I saw this elsewhere quite a lot)